torch/autograd/saved_variable_default_hooks.py

```diff
@@ -5,3 +5,19 @@ def set_saved_tensors_default_hooks(pack_hook, unpack_hook):
 
 def reset_saved_tensors_default_hooks():
     torch._C._autograd._reset_default_hooks()
+
+def set_save_on_cpu_hooks():
+    def pack_hook(tensor):
+        storage = torch.empty(
+            tensor.size(),
+            dtype=tensor.dtype,
+            layout=tensor.layout,
+            pin_memory=torch.cuda.is_available())
```

Comment:
What is the rationale for using pin_memory? That the transfers will be faster?

The problem with pinned memory is that there is generally a limited amount of it and for larger models we'll probably run out of pinned memory and raise an error here. A good sanity check might be something like running resnet18 training

Reply:
Yes. It's so that we can use a non-blocking copy from CPU to GPU. See https://github.com/pytorch/pytorch/pull/61928#issuecomment-887009448 for an example on resnet152.

One option I like would be to take as argument a `pin_memory` bool so the user can decide whether they want pinned memory or not.

Reply:
Conventional wisdom tells me that `pin_memory` should not be used by default because it is a limited resource. I agree a pin_memory book would be nice for the user to decide if they want pinned memory or not